File: cs/DressDiscover/DressDiscover.Api/src/DressDiscover/Api/Models/Worksheet/WorksheetFeatureId.cs
namespace DressDiscover.Api.Models.Worksheet
{
    public sealed class WorksheetFeatureId {
        public static WorksheetFeatureId Parse(string value) {
            if (value.Length == 0) {
                throw new System.ArgumentException("id is empty", "WorksheetFeatureId");
            }
            return new WorksheetFeatureId(value);
        }

        private WorksheetFeatureId(string value) {
            this.value = value;
        }

        public override string ToString() {
            return this.value;
        }

        private string value;
    }
}

File: worksheet/ts/src/ts/models/worksheet/WorksheetFeatureId.ts
export class WorksheetFeatureId {
  public static parse(id: string): WorksheetFeatureId {
    if (id.trim().length === 0) {
      throw new RangeError("id is empty");
    }
    return new WorksheetFeatureId(id);
  }

  public equals(other?: WorksheetFeatureId): boolean {
    if (typeof other === "undefined") {
      return false;
    }
    return this.id === other.id;
  }

  public toString(): string {
    return this.id;
  }

  private constructor(private readonly id: string) {}
}
